test/unit/shadow/polyfill.js

```diff
@@ -38,6 +39,21 @@ describe('shadow/polyfill', function () {
     expect(root.tagName).to.equal('TEST-ING');
   });
 
+  if (canPatchNativeAccessors) {
+    it('proper node removal', function () {
+      const host = create('div');
+      host.appendChild(create('div'));
+      host.appendChild(create('div'));
+      host.appendChild(create('div'));
+      host.appendChild(create('div'));
+      host.attachShadow({ mode: 'open' });
+
+      expect(host.__childNodes.length).to.equal(1);
+      expect(host.__childNodes[0].tagName).to.equal('_SHADOW_ROOT_');
```

Comment:
A more robust check would be to do:

```js
expect(host.__childNodes[0]).to.equal(host.shadowRoot);
```

Or, you could use the return value of `attachShadow()` as the `shadowRoot`.

The only reason I pose this is if the name of the shadow root tag name changes then the test breaks.